src/werkzeug/formparser.py

```diff
@@ -461,6 +469,160 @@ def validate_boundary(self, boundary):
             # the assert is skipped.
             self.fail("Boundary longer than buffer size")
 
+    def _split_lines(self, input_buffer, cap=None):
+        lines = input_buffer.splitlines(True)
+        for line in lines:
+            if cap:
+                for low_bound in range(0, len(line), cap):
+                    yield line[low_bound : low_bound + cap], b""
+                remainder = len(line) % cap
+                line = line[-remainder:]
```

Comment:
You will end up missing parts of the line.

`line` is your loop variable, so reassigning it won't help much.

If you that behavior, you'll probably need a deque, somethine like

```
lines = input_buffer.splitlines(True)
lines = collections.deque(lines)
while lines:
  line = lines.popleft()
  if cap:
    [...]
    remainder =len(line) % cap
    lines.appendleft(line[-remainder:])
    [...]
```